open-apparel-registry/open-apparel-registry
# Add linking IDs for relocated facilities

## Overview

Sometimes facilities close and relocate to a new address. In these
instances, the original facility will be marked as closed, but we
should be able to link the closed facility to the new facility ID.

Unlike deleted facilities, users should still be able to see the closed
facility's details, so they are not rerouted automatically. Instead, on
a closed facility profile, a link is added to allow users to easily
navigate to the new facility details page.

In addition, administrators require a method to link the closed
facilities to the new ids. Therefore a new page has been added to the
admin dashboard allowing a new OAR ID to be linked to a facility.

Connects #1215 

## Demo

<img width="1387" alt="Screen Shot 2021-09-27 at 2 31 30 PM" src="https://user-images.githubusercontent.com/21046714/135087671-cb08560c-34f4-4885-ac0f-129ff8317f30.png">
<img width="1385" alt="Screen Shot 2021-09-27 at 3 38 14 PM" src="https://user-images.githubusercontent.com/21046714/135087685-55903cb0-4907-422a-aec4-c3720152d958.png">
<img width="1124" alt="Screen Shot 2021-09-27 at 3 38 28 PM" src="https://user-images.githubusercontent.com/21046714/135087694-d92895d2-a664-4485-82f7-e8b1eac6e331.png">

## Testing Instructions

* Run `./scripts/manage migrate`
* Run `./scripts/server`
* Close a facility, and note the ID of the closed facility. 
* Obtain another facility ID. Navigate to the admin dashboard and to Link to New OAR ID. 
* Enter the closed facility ID and click search - the facility should be found. 
* Enter the other facility ID and click search - the facility should be found. 
* Click 'link'. You should see a label by the closed facility listing the new OAR ID. 
* Navigate to the closed facility details page. You should see a Closed banner with a link to the other facility. 
* Fetch the history for the closed facility from the API. You should see the new OAR ID change listed. 

## Checklist
